llvm/lib/SYCLLowerIR/SYCLCreateNVVMAnnotations.cpp

```diff
@@ -17,17 +17,18 @@
 
 using namespace llvm;
 
-static void addNVVMMetadata(GlobalValue &GV, StringRef Name, int Operand) {
-  Module *M = GV.getParent();
+static void addNVVMMetadata(Function &F, StringRef Name, int Operand) {
+  Module *M = F.getParent();
   LLVMContext &Ctx = M->getContext();
 
   // Get "nvvm.annotations" metadata node
   NamedMDNode *MD = M->getOrInsertNamedMetadata("nvvm.annotations");
 
-  Metadata *MDVals[] = {ConstantAsMetadata::get(&GV), MDString::get(Ctx, Name),
+  Metadata *MDVals[] = {ConstantAsMetadata::get(&F), MDString::get(Ctx, Name),
                         ConstantAsMetadata::get(
                             ConstantInt::get(Type::getInt32Ty(Ctx), Operand))};
   // Append metadata to nvvm.annotations
+  F.setCallingConv(CallingConv::PTX_Kernel);
```

Comment:
I'm not sure this is necessary, or if it is it's being done in the wrong place. The `SYCLCreateNVVMAnnotations` doesn't "create kernels", it only converts (non-kernel) annotations from metadata.

Could we try without this change and try and understand what's going on, if there are any fails?